pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/SegmentPreloadUtils.java

```diff
@@ -191,4 +199,33 @@ private static Map<String, SegmentZKMetadata> getSegmentsZKMetadata(String table
         .forEach(m -> segmentMetadataMap.put(m.getSegmentName(), m));
     return segmentMetadataMap;
   }
+
+    /**
+     * Get a virtual data source for the given column in the table.
+     *
+     *
+     * @param column Column name for which the data source is needed
+     * @param totalDocCount Total document count for the column
+     * @return DataSource for the column
+     */
+  public static DataSource getVirtualDataSource(Schema tableSchema, String column, int totalDocCount) {
```

Comment:
This method doesn't belong to this util class. The virtual `DataSource` is created at query time, but not load time. Consider adding a new util class for it